src/main/java/com/orderService/OrderService/service/OrderServiceImpl.java

```diff
@@ -56,14 +62,13 @@ public Order createOrder(OrderDTO orderDTO) throws ExecutionException, Interrupt
             throw new InvalidType("Either user or product not found. Detailed message : " + exception.getMessage());
         }
         if (user.getUserType() == UserType.MAGIC && product.getProductType() == ProductType.MAGIC)
```

Comment:
I think all these `if` are logic around how to construct of the object, so should be inside the orderFactory which is a creational design pattern